src/main/java/org/candlepin/service/impl/HypervisorUpdateAction.java

```diff
@@ -134,10 +136,15 @@ public Result update(
         return new Result(result, hypervisorConsumersMap);
     }
 
-    public Consumer reconcileHost(Owner owner, Consumer incomingHost, HypervisorUpdateResultDTO result,
+    public Consumer reconcileHost(Owner owner, ConsumerDTO incomingHost, HypervisorUpdateResultDTO result,
         boolean create, String principal, String jobReporterId) {
-        String systemUuid = incomingHost.getFact(Consumer.Facts.DMI_SYSTEM_UUID);
-        String hypervisorId = incomingHost.getHypervisorId().getHypervisorId();
+        String systemUuid = incomingHost.getFacts() == null ? null :
+            incomingHost.getFacts().get(Consumer.Facts.DMI_SYSTEM_UUID);
+        // When the data comes in via a HypervisorIdDTO the lowercasing is not applied as per
+        // the HypervisorId class method setHypervisorId. The hypervisor id lookup is the only
+        // place where it will matter as the DTO data will be put through the model object before
+        // any other operations.
+        String hypervisorId = incomingHost.getHypervisorId().getHypervisorId().toLowerCase();
```

Comment:
Would it be better to apply the `toLowerCase()` in the getExistingConsumerByHypervisorIdOrUuid method? I imagine that this can be more helpful so that in the future if getExistingConsumerByHypervisorIdOrUuid is used somewhere else, the implementer doesn't have to worry about applying the `toLowerCase()` to the hypervisorId value before using the method. Also, it looks like this is already done for the systemUuid value in that method.

Reply:
Good point